test/Libraries/DynamoPythonTests/PythonEvalTestsWithLibraries.cs

```diff
@@ -43,5 +44,164 @@ import clr
+# native = array('l', [1,2])
+# native = List.AddItemToEnd(3, native) 
+
+# .NET array => Python list - Works in both engines
+a = DummyCollection.MakeArray(1,2)
+a[0] = a[1] + 1
+b = len(a)
+a[1] = b
+a = List.AddItemToEnd(1, a)
+
+OUT = a
+";
+            var empty = new ArrayList();
+            var expected = new ArrayList { 3, 2, 1 };
+            foreach (var pythonEvaluator in Evaluators)
+            {
+                var result = pythonEvaluator(code, empty, empty);
+                Assert.IsTrue(result is IEnumerable);
+                CollectionAssert.AreEqual(expected, result as IEnumerable);
+            }
+        }
+
+        [Test]
+        [Category("Failure")]
+        public void TestTupleEncoding()
+        {
+            string code = @"
+import sys
+import clr
+clr.AddReference('FFITarget')
+clr.AddReference('DSCoreNodes')
+from FFITarget import DummyCollection
+from DSCore import List
+
+t = (1,2,3)
+# Python tuple => .NET array - Works in both
+a = DummyCollection.MakeArray(t)
+# Python tuple => .NET IList - Does not work in CPython
+l = List.AddItemToEnd(4, t)
+
+OUT = a, l
+";
+            var empty = new ArrayList();
+            var expected = new ArrayList { new ArrayList { 1, 2, 3 }, new ArrayList { 1, 2, 3, 4 } };
+            foreach (var pythonEvaluator in Evaluators)
+            {
+                var result = pythonEvaluator(code, empty, empty);
+                Assert.IsTrue(result is IEnumerable);
+                CollectionAssert.AreEqual(expected, result as IEnumerable);
+            }
+        }
+
+        [Test]
+        [Category("Failure")]
+        public void TestRangeEncoding()
+        {
+            string code = @"
+import sys
+import clr
+clr.AddReference('FFITarget')
+clr.AddReference('DSCoreNodes')
+from FFITarget import DummyCollection
+from DSCore import List
+
+r = range(0, 10, 2)
+# Python range => .NET array - Works in both
+a = DummyCollection.MakeArray(r)
+# Python range => .NET IList - Does not work in CPython
+l = List.AddItemToEnd(10, r)
+
+OUT = a, l
+";
+            var empty = new ArrayList();
+            var expected = new ArrayList { new ArrayList { 0, 2, 4, 6, 8 }, new ArrayList { 0, 2, 4, 6, 8, 10 } };
+            foreach (var pythonEvaluator in Evaluators)
+            {
+                var result = pythonEvaluator(code, empty, empty);
+                Assert.IsTrue(result is IEnumerable);
+                CollectionAssert.AreEqual(expected, result as IEnumerable);
+            }
+        }
+
+        [Test]
+        [Category("Failure")]
+        public void TestDictionaryEncoding()
+        {
+            string code = @"
+import sys
+import clr
+clr.AddReference('FFITarget')
+clr.AddReference('DesignScriptBuiltin')
+from DesignScript.Builtin import Dictionary
+from FFITarget import DummyCollection
+
+d = {'one': 1, 'two': 2, 'three': 3}
+# Python dict => DS Dictionary - Does not work in either engine
```

Comment:
I'm not sure if this is intended. It would be good to have but if it were to be supported, we would most likely need to make custom changes in pythonnet for interop with DS that would not be accepted by the main repo for back-porting and we would need to continue to maintain our own custom fork.

Reply:
We didn't do it for IronPython either.

Reply:
Thanks for the feedback. I thought of a way to do this without referencing DS Dictionary in Python.NET. What we would do is a two step conversion:
1. Convert it to a .NET Dictionary, using primitives from Python.NET
2. Convert the .NET Dictionary to a DS Dictionary, as part of a custom encoder in Dynamo